tests/MongoDB.Driver.Core.Tests/Core/Helpers/MockClusterableServerFactory.cs

```diff
@@ -118,13 +115,15 @@ public void PublishDescription(ServerDescription description)
                 var mockServer = Mock.Get(result.Server);
                 mockServer.SetupGet(s => s.Description).Returns(description);
                 mockServer.Raise(s => s.DescriptionChanged += null, new ServerDescriptionChangedEventArgs(oldDescription, description));
-
             }
             else
             {
                 var mockMonitor = Mock.Get(result.Monitor);
                 mockMonitor.SetupGet(m => m.Description).Returns(description);
-                mockMonitor.Raise(m => m.DescriptionChanged += null, new ServerDescriptionChangedEventArgs(oldDescription, description));
+                if (ServerMonitorReflector.ShouldDescriptionChangedEventBePublished(oldDescription, description))
```

Comment:
Since `result.Monitor` is only interface, I created `ShouldDescriptionChangedEventBePublished` as a static method to be able to call it here